SYCL/Matrix/joint_matrix_tensorcore.cpp

```diff
@@ -114,99 +121,151 @@ void test() {
+                                                                       cgh);
+
+        cgh.parallel_for<KernelName<bfloat16, class copyA, M, K, N>>(
+            range<1>(Big_M * Big_K), [=](item<1> item) {
+              auto i = item.get_linear_id();
+              accA[i] = 0.1f * (i % 10);
+            });
+      });
+
+      q.submit([&](handler &cgh) {
+        accessor<T1, 1, access::mode::read_write, target::device> accB(bufB,
+                                                                       cgh);
+
+        cgh.parallel_for<KernelName<bfloat16, class copyB, M, K, N>>(
+            range<1>(Big_K * Big_N), [=](item<1> item) {
+              auto i = item.get_linear_id();
+              accB[i] = 0.1f * (i % 10);
+            });
+      });
+    }
+
+    q.submit([&](handler &cgh) {
+      accessor<T1, 1, access::mode::read_write, target::device> accA(bufA, cgh);
+      accessor<T1, 1, access::mode::read_write, target::device> accB(bufB, cgh);
+      accessor<T2, 1, access::mode::read_write, target::device> accC(bufC, cgh);
+      accessor<T2, 1, access::mode::read_write, target::device> accD(bufD, cgh);
+
+      range<2> LocalRange = {1, N_THREADS_PER_MATRIX_OP};
+      range<2> GlobalRange = {Sub_Tiles_M,
+                              Sub_Tiles_N * N_THREADS_PER_MATRIX_OP};
+
+      cgh.parallel_for<KernelName<T1, T2, M, K, N>>(
+          nd_range<2>(GlobalRange, LocalRange), [=](nd_item<2> item) {
+            sub_group sg = item.get_sub_group();
+            const auto m =
+                item.get_group().get_group_id()[0]; // row id of current
+                                                    // submatrix of BIG C matrix
+            const auto n =
+                item.get_group().get_group_id()[1]; // column id of current
+                                                    // submatrix of BIG C matrix
+
+            joint_matrix<T1, matrix_use::a, M, K, matrix_layout::row_major>
+                sub_a;
+
+            joint_matrix<T1, matrix_use::b, K, N, matrix_layout::row_major>
+                sub_b;
+
+            joint_matrix<T2, matrix_use::accumulator, M, N,
+                         matrix_layout::row_major>
+                sub_c;
+
+            joint_matrix_load(
+                sg, sub_c, accC.get_pointer() + (m * M) * Big_N + n * N, Big_N);
+
+            for (int k = 0; k < Sub_Tiles_K;
+                 k++) // row/col id of current submatrix of BIG A/B matrices
+            {
+              joint_matrix_load(sg, sub_a,
+                                accA.get_pointer() + (k * K) + (m * M * Big_K),
+                                Big_K);
+
+              if constexpr (std::is_same<T1, bfloat16>::value) {
+                marray<bfloat16, sub_a.wi_marray.size()> b, c;
+                b = 2;
+                c = 1;
+                sub_a.wi_marray =
+                    sycl::ext::oneapi::experimental::fma(sub_a.wi_marray, b, c);
```

Comment:
Is this supported for int8_t and half as well?

Reply:
full sycl 2020 marray support including all genfloat types: half, float, double has been added in the open PR here: https://github.com/intel/llvm/pull/6038. integer types are not supported by this function.